client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -1115,6 +1128,11 @@ const ConnectedPlanFeatures2023Grid = connect(
 
 			const availableForPurchase = isInSignup || isPlanAvailableForPurchase( state, siteId, plan );
 			const isCurrentPlan = isCurrentSitePlan( state, siteId, planProductId ) ?? false;
+			let isVisible = visiblePlans.indexOf( plan ) !== -1;
+
+			if ( hideFreePlan && plan === PLAN_FREE ) {
+				isVisible = false;
+			}
```

Comment:
Moved the `hideFreePlan` here so that all visibility computation is in one place.